doc/faq/virtualenv_faq.rst

```diff
@@ -0,0 +1,106 @@
+.. _virtualenv-faq:
+
+***********************************************
+Working with Matplotlib in Virtual environments
+***********************************************
+
+.. contents::
+   :backlinks: none
+
+
+.. _introduction:
+
+Introduction
+============
+
+When running :mod:`matplotlib` in a virtual environment you may discover a
+few issues. :mod:`matplotlib` itself has no issue with virtual environments.
+However, the GUI frameworks that :mod:`matplotlib` uses for interactive
+figures have some issues with virtual environments. Everything below assumes
+some familiarity with the Matplotlib backends as found in :ref:`What is a
+backend? <what-is-a-backend>`.
+
+If you only use the ``IPython/Jupyter Notebook``'s ``inline`` and ``notebook``
+backends and non interactive backends you should not have any issues and can
+ignore everything below.
+
+GUI Frameworks
+==============
+
+Interactive Matplotlib relies heavily on the interaction with external GUI
+frameworks.
+
+Most GUI frameworks are not pip installable. This makes it tricky to install
+them within a virtual environment. This problem does not exist if you use Conda
+environments where you can install all Conda supported GUI frameworks directly
+into the environment. In regular virtualenv environment various workarounds
+exist. Some of these are given here:
+
+The ``TKAgg`` backend doesn't require any external dependencies and is normally
```

Comment:
Maybe make these separate bullet points, one for TkAgg and one for PySide, since I think if you were skimming this for info about a particular backend, you may miss PySide.